src/Utils/SourceGenerator/readme.md

```diff
@@ -0,0 +1,174 @@
+Following the suggestions of Mukesh
+
+Should be under src/Utils/SourceGenerator
+with assembly name as FSH.WebApi.SourceGenerator
+
+The code is based on the product and brand entities
+I used the same aproach as I did some years ago so bear with me should the whole thing is well 'not that modern' .
+But at the end of the 'run' all the files are created.
+After a rerun fullstackhero sets the permissions (as it does now) .
+
+with the new directories in place (and if not completely integrated in fullstackhero)
+Add a reference to the Utils project in the host project
+
+in program.cs
+Add 'using FSH.WebApi.Utils;'
+Add 'builder.Services.AddUtils(builder.Configuration);'
+Add 'await app.Services.InitializeGenerateSources();' ' 
```

Comment:
I don't really understand why you would want the source generation in the host project? Surely source generation is something you would only want to do at development time... never at runtime?

I would think the source generator itself would be a cli utility (could even make an actual dotnet tool out of if)... which takes some parameters as to what to generate?